oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/userinfo/OidcUserService.java

```diff
@@ -127,8 +131,9 @@ public OidcUser loadUser(OidcUserRequest userRequest) throws OAuth2Authenticatio
 			}
 		}
 
-		Set<GrantedAuthority> authorities = Collections.singleton(
-				new OidcUserAuthority(userRequest.getIdToken(), userInfo));
+		Set<GrantedAuthority> authorities = new LinkedHashSet<>();
+		authorities.add(new OidcUserAuthority(userRequest.getIdToken(), userInfo));
+		authorities.addAll(oauth2UserAuthorities);
```

Comment:
The tests already enforce this -- see that they check the size of the returned authorities as well as the types of the elements.

Because of the `DefaultOAuth2User#sortAuthorities` method, the duplicate is eliminated.

Is there anything additional you'd like to see to handle this?